dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -211,18 +206,18 @@ protected static String extractDoiFromDoiUrl(String doiUrl) {
     /**
      * Create a workflow alias that uses a digital object identifier
      * @param doi digital object identifier
-     * @param workflow workflow for which DOI is registered (the list
-     *                 of aliases in the workflow is updated as a side
-     *                 effect)
+     * @param workflowResource code for interacting with the files of versions(we use zip file creation methods) and aliases
+     * @param user user authenticated to issue a DOI for the workflow
      * @return the alias as a string
      */
-    private static String createAliasUsingDoi(String doi, Workflow workflow) {
-        Map<String, Alias> aliases = workflow.getAliases();
+    protected static String createAliasUsingDoi(String doi, WorkflowResource workflowResource, User user) {
         // Replace forward slashes so we can use the DOI in an alias
         String doiReformattedAlias = doi.replaceAll("/", "-");
-        // This adds an alias to the workflow's list of aliases
-        // so it updates workflow as a side effect
-        aliases.put(doiReformattedAlias, new Alias());
+        // Make sure the alias is valid
+        // If it is not acceptable then an exception is generated
+        Set<String> aliasSet = new HashSet<>();
+        aliasSet.add(doiReformattedAlias);
```

Comment:
You can optionally use `Collections.singleton` instead.

Reply:
done